src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -1068,21 +1068,54 @@ private async Task SendInviteAsync(OrganizationUser orgUser, Organization organi
             IUserService userService)
         {
             var orgUser = await _organizationUserRepository.GetByIdAsync(organizationUserId);
+            
             if (orgUser == null)
             {
                 throw new BadRequestException("User invalid.");
             }
-
-            if (orgUser.Status != OrganizationUserStatusType.Invited)
+            
+            if (!CoreHelpers.UserInviteTokenIsValid(_dataProtector, token, user.Email, orgUser.Id, _globalSettings))
             {
-                throw new BadRequestException("Already accepted.");
+                throw new BadRequestException("Invalid token.");
             }
-
+            
             if (string.IsNullOrWhiteSpace(orgUser.Email) ||
                 !orgUser.Email.Equals(user.Email, StringComparison.InvariantCultureIgnoreCase))
             {
                 throw new BadRequestException("User email does not match invite.");
             }
+            
+            var existingOrgUserCount = await _organizationUserRepository.GetCountByOrganizationAsync(
+                orgUser.OrganizationId, user.Email, true);
+            if (existingOrgUserCount > 0)
+            {
+                throw new BadRequestException("You are already part of this organization.");
+            }
+            
+            return await AcceptUserAsync(orgUser, user, userService);
+        }
+        
+        public async Task<OrganizationUser> AcceptUserAsync(string orgIdentifier, User user, IUserService userService)
+        {
+            var org = await _organizationRepository.GetByIdentifierAsync(orgIdentifier);
```

Comment:
Do you need to null check some things here? What if org is null?

Reply:
Good catch - added null check for `org`